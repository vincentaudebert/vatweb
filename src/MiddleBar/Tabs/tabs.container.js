import { connect } from 'react-redux';
import Tabs from './tabs';
import { closeFile } from '../../App/app.actions';
import { withRouter } from 'react-router-dom';

const mapStateToProps = (state, ownProps) => {
  return {
    openFiles: state.openFiles,
  };
};

const mapDispatchToProps = (dispatch, ownProps) => ({
  closeFile: (openFiles, slug) => {
    dispatch(closeFile(slug));
  },
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(Tabs)
);
