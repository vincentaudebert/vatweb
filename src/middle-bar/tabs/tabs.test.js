import React from 'react';
import renderer from 'react-test-renderer';
/* eslint-disable import/no-extraneous-dependencies */
import { MemoryRouter } from 'react-router';
/* eslint-enable import/no-extraneous-dependencies */
import { mount } from 'enzyme';
import Tabs from './tabs';

const routerWrapper = explorer => (
  <MemoryRouter keyLength={0}>{explorer}</MemoryRouter>
);

describe('Tabs', () => {
  let openFiles;
  const closeFile = jest.fn();

  beforeEach(() => {
    openFiles = [
      {
        name: 'welcome.json',
        active: true,
      },
      {
        name: 'me.js',
        active: false,
      },
      {
        name: '1-apocope.js',
        active: false,
      },
    ];
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        routerWrapper(<Tabs openFiles={openFiles} closeFile={closeFile} />)
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with no open files', () => {
    openFiles = [];
    const tree = renderer
      .create(
        routerWrapper(<Tabs openFiles={openFiles} closeFile={closeFile} />)
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('close file button works', () => {
    const tabs = mount(
      routerWrapper(<Tabs openFiles={openFiles} closeFile={closeFile} />)
    );
    tabs
      .find('button')
      .first()
      .simulate('click');

    expect(closeFile).toHaveBeenCalled();
  });

  it('filter out invalid openFiles', () => {
    openFiles[0].name = 'invalid name';
    const tabs = mount(
      routerWrapper(<Tabs openFiles={openFiles} closeFile={closeFile} />)
    );
    const tabItems = tabs.find('div.tab');
    expect(tabItems.length).toBe(2);
    expect(tabs).toMatchSnapshot();
  });
});
