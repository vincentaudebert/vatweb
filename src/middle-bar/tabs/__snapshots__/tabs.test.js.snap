// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs filter out invalid openFiles 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Tabs
      closeFile={
        [MockFunction] {
          "calls": Array [
            Array [
              Array [
                Object {
                  "active": true,
                  "content": "{
  \\"hello\\": [
    \\"Heya! Welcome on my portfolio.\\", 
    \\"It shows you what I have done and what I can do.\\"
  ],
  \\"wait-what\\": [
    \\"Ok, maybe it needs a bit of explanation.\\",
    \\"You can use the file explorer on your left to open files\\",
    \\"These describe some of the work I have been doing or what studies I did, etc.\\"
  ],
  \\"are-you-mad\\": [
    \\"Hehe, no. I just thought it would be\\",
    \\"an original way to present myself and relevant with my expertise.\\"
  ],
  \\"aint-nobody-got-time-for-that\\": [
    \\"Ok if you can't be bothered\\",
    \\"you can download directly my CV in a PDF format\\",
    \\"by clicking on the PDF icon in the menu bar on the left (desktop only)\\",
    \\"or click on the smiley face in the footer bar.\\"
  ],
  \\"where-is-da-code\\": [
    \\"Oh by the way, I made the source code open source.\\",
    \\"You can see it here:\\",
    \\"https://github.com/vincentaudebert/vatweb\\",
    \\"or click on the git icon in the footer bar.\\",
  ],
  \\"hey-granpa\\": [
    \\"Yes. I know. This code is fairly old and a bit outdated. Not even using React hooks...\\",
    \\"And it's only from june 2018!\\",
    \\"But it means React is moving fast! So we can't really complain :)\\",
    \\"I will update it ASAP when I get some spare time\\",
  ]
}",
                  "inFolder": false,
                  "language": "json",
                  "name": "welcome.json",
                  "type": "file",
                },
                Object {
                  "active": false,
                  "content": "import React, { Component } from 'react';
import type { Props } from './home.flow.js';

class Home extends Component<Props, *> {
  constructor(props: Props) {
    super(props);
    const { setBirthdayDate, setBirthdayLocation } = props;
    setBirthdayDate('5 may 1989');
    setBirthdayLocation('Athis-mons, France');
  }

  componentDidMount() {
    const {
      setLocation,
      setPhoneNumber,
      setEmail,
      setLinkedInProfile,
      setLanguages
    } = this.props;
    setLocation('Bonneville, France');
    setPhoneNumber('+33 7 72 41 88 58');
    setEmail('vincent@vatweb.fr');
    setLinkedInProfile('https://www.linkedin.com/in/vincentaudebert/');
    setGitHubProfile('https://github.com/vincentaudebert');
    setLanguages(['French', 'English']);
  }

  componentWillUnmount() {
    const { useOneLife, transformInto } = this.props;
    useOneLife();
    transformInto('eagle');
  }

  render() {
    return (
      <div className=\\"summary\\">
        <p>
          I'm a front-end developer with more than 6 years of work experience in
          web development.
        </p>
        <p>
          I have worked in several projects as a web developer but also as an
          assistant project manager. This gives me a good knowledge of every
          aspect of a project and of what can make it a success.
        </p>
        <p>
          I have a strong passion to create cutting-edge and high quality
          websites, and the technical web development experience required to
          make it happen.
        </p>
        <p>I am working fully remotely since June 2017.</p>
      </div>
    );
  }
}

export default Home;",
                  "inFolder": false,
                  "language": "react",
                  "name": "me.js",
                  "type": "file",
                },
                Object {
                  "active": false,
                  "content": "import React, { Component, Fragment } from 'react';
import type { Props } from './apocope.flow.js';

class Apocope extends Component<Props, *> {
  constructor(props: Props) {
    const { setStartDate, setWorkLocation } = props;
    setStartDate('September 2008');
    setWorkLocation('Paris, France');
  }

  componentDidMount() {
    const { setSkills } = this.props;
    setSkills([
      'PHP',
      'Zend Framework',
      'Responsive design',
      'HTML5',
      'Unit tests',
      'R&D',
      'Geolocation',
    ]);
  }

  componentWillUnmount() {
    const { setEndDate } = this.props;
    setEndDate('September 2010');
  }

  render() {
    return (
      <Fragment>
        <p>
          During this tenure, I was an apprentice, working primarily as a web
          developer on many mobile website projects.
        </p>

        <div className=\\"accomplishments\\">
          <ul>
            <li>
              <h3>Conception of mobile internet websites</h3>
              <p>Nespresso</p>
              <p>Carrefour</p>
              <p>Decathlon</p>
              <p>Nespresso</p>
            </li>
            <li>
              <h3>Research and Development on the company platform</h3>
              <p>
                Development of a geolocalization module based on Google Maps
              </p>
              <p>Integration of Zend to the company platform</p>
              <p>Study of HTML5</p>
            </li>
            <li>Unit tests</li>
            <li>Preparation of technical specifications</li>
          </ul>
        </div>
      </Fragment>
    );
  }
}

export default Apocope;
",
                  "inFolder": true,
                  "language": "react",
                  "name": "1-apocope.js",
                  "type": "file",
                },
              ],
              "welcome.json",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      openFiles={
        Array [
          Object {
            "active": true,
            "name": "invalid name",
          },
          Object {
            "active": false,
            "name": "me.js",
          },
          Object {
            "active": false,
            "name": "1-apocope.js",
          },
        ]
      }
    >
      <div
        className="tabs"
      >
        <div
          className="tab d-inline-block"
          key="tabs-me.js"
        >
          <Link
            className="file--link"
            to="/file/me.js"
          >
            <LinkAnchor
              className="file--link"
              href="/file/me.js"
              navigate={[Function]}
            >
              <a
                className="file--link"
                href="/file/me.js"
                onClick={[Function]}
              >
                <img
                  alt="Type Icon"
                  className="seti-icon seti-icon--big"
                  src="react.svg"
                />
                 
                me.js
                 
                <button
                  className="reset-button"
                  onClick={[Function]}
                  title="Close me.js"
                  type="button"
                >
                  <MdClose
                    className="icon--tab"
                  >
                    <IconBase
                      attr={
                        Object {
                          "viewBox": "0 0 24 24",
                        }
                      }
                      className="icon--tab"
                    >
                      <svg
                        className="icon--tab"
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        strokeWidth="0"
                        style={
                          Object {
                            "color": undefined,
                          }
                        }
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                          key="0"
                        />
                      </svg>
                    </IconBase>
                  </MdClose>
                </button>
              </a>
            </LinkAnchor>
          </Link>
        </div>
        <div
          className="tab d-inline-block"
          key="tabs-1-apocope.js"
        >
          <Link
            className="file--link"
            to="/file/1-apocope.js"
          >
            <LinkAnchor
              className="file--link"
              href="/file/1-apocope.js"
              navigate={[Function]}
            >
              <a
                className="file--link"
                href="/file/1-apocope.js"
                onClick={[Function]}
              >
                <img
                  alt="Type Icon"
                  className="seti-icon seti-icon--big"
                  src="react.svg"
                />
                 
                1-apocope.js
                 
                <button
                  className="reset-button"
                  onClick={[Function]}
                  title="Close 1-apocope.js"
                  type="button"
                >
                  <MdClose
                    className="icon--tab"
                  >
                    <IconBase
                      attr={
                        Object {
                          "viewBox": "0 0 24 24",
                        }
                      }
                      className="icon--tab"
                    >
                      <svg
                        className="icon--tab"
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        strokeWidth="0"
                        style={
                          Object {
                            "color": undefined,
                          }
                        }
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                          key="0"
                        />
                      </svg>
                    </IconBase>
                  </MdClose>
                </button>
              </a>
            </LinkAnchor>
          </Link>
        </div>
      </div>
    </Tabs>
  </Router>
</MemoryRouter>
`;

exports[`Tabs renders correctly 1`] = `
<div
  className="tabs"
>
  <div
    className="tab d-inline-block tab--active"
  >
    <a
      className="file--link"
      href="/file/welcome.json"
      onClick={[Function]}
    >
      <img
        alt="Type Icon"
        className="seti-icon seti-icon--big"
        src="json.svg"
      />
       
      welcome.json
       
      <button
        className="reset-button"
        onClick={[Function]}
        title="Close welcome.json"
        type="button"
      >
        <svg
          className="icon--tab"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </button>
    </a>
  </div>
  <div
    className="tab d-inline-block"
  >
    <a
      className="file--link"
      href="/file/me.js"
      onClick={[Function]}
    >
      <img
        alt="Type Icon"
        className="seti-icon seti-icon--big"
        src="react.svg"
      />
       
      me.js
       
      <button
        className="reset-button"
        onClick={[Function]}
        title="Close me.js"
        type="button"
      >
        <svg
          className="icon--tab"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </button>
    </a>
  </div>
  <div
    className="tab d-inline-block"
  >
    <a
      className="file--link"
      href="/file/1-apocope.js"
      onClick={[Function]}
    >
      <img
        alt="Type Icon"
        className="seti-icon seti-icon--big"
        src="react.svg"
      />
       
      1-apocope.js
       
      <button
        className="reset-button"
        onClick={[Function]}
        title="Close 1-apocope.js"
        type="button"
      >
        <svg
          className="icon--tab"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </button>
    </a>
  </div>
</div>
`;

exports[`Tabs renders correctly with no open files 1`] = `
<div
  className="tabs"
/>
`;
